pulpsolv.py

```diff
@@ -286,11 +299,47 @@ class RpmUnitSolvableFactory(BasetUnitSolvableFactory):
     ]
 
 
-def load_repo_units(plugin_manager, repo_name, factory):
+class ErratumSolvableFactory(BasetUnitSolvableFactory):
+    attribute_factories = [
+        # cargo-culting from
+        # https://github.com/openSUSE/libsolv/blob/master/ext/repo_updateinfoxml.c
+        AttributeFactory('errata_id', target_attr='name',
+                         conversion=lambda x: str('patch:' + x)),
+        AttributeFactory('arch', default='noarch'),
+        AttributeFactory('errata_from', target_attr='vendor'),
+        EVRAttributeFactory(),
+        # FIXME: not all these units are really required; what pulp does is it
+        # filters rpm_search_dicts to nevras actually present in the source repo:
+        #   https://github.com/pulp/pulp_rpm/blob/ef5fc5b2af47736114b68bc08658d9b2a94b84e1/plugins/pulp_rpm/plugins/importers/yum/associate.py#L91,#L94
+        # Could this be handled as some ignore_missing?
+        RpmDependencyAttributeFactory(
+            'rpm_search_dicts', dependency_key=solv.SOLVABLE_REQUIRES),
+    ]
+
+    def __call__(self, unit):
+        # FIXME: somehow, an erratum has to explicitly provide itself; see also:
```

Comment:
this is expected. libsolv doesn't do self-provides. You have to be explicit

Reply:
cool, somehow it works for RPMs; should I add that there too?